### ISSUE TITLE
FactPEGASUS: Factuality-Aware Pre-training and Fine-tuning for Abstractive Summarization

### ISSUE THREAD

**Issue by ichiroex** [ichiroex/nlp-papers#15]
> Wan, David, and Mohit Bansal. 2022. “FactPEGASUS: Factuality-Aware Pre-Training and Fine-Tuning for Abstractive Summarization.” arXiv [cs.CL]. arXiv. http://arxiv.org/abs/2205.07830.

## 解決したい課題
- 事前学習モデルでは、事前学習時にsource-target pairのfactualityを考慮できていない
  - 多くの研究ではfine-tuning時にだけfactualityを考慮している

## 提案手法
- (1) PEGASUSのpretraining objectiveを拡張して、source-targer pairを作成する方法を提案する
- (2) Fine-tuningのために3つのコンポーネントを提案する
  - corrector: reference summaryからhallucination（entity）を削除する
  - contrastor:  contrastive learningにより、nonfactual summaryとfactual summaryをモデルが区別できるように学習する
  - connector: pre-trainingとfine-tuningのgapを解消する方法（fine-tuning時にも[MASK]トークンを導入する）

<img width="1162" alt="スクリーンショット 2022-08-16 10 12 31" src="https://user-images.githubusercontent.com/13919583/184773937-24e6087c-f3e5-4537-8ac2-572036267942.png">

### Factuality-Aware Pre-training

- gap sentence generation（GSG）
  - ソース文書を擬似的な要約データに変換する方法（PEGASUSで使われている手法）
  - ROUGEスコアを使って、ソース文書から要約文を抽出する
  - 各文とそれ以外の文書のROUGE-1を算出、スコアが高いもの要約文として選択したい
  - ソース文書のうち、30%を選択
  - しかし、選択された文がfaithfulかどうかは保証できない（選択された文はmaskされるので）
- 提案: factGSG
  - hallucinatiuonの可能性を低減するために、ソース文書のうち、1文を選択
  - FactCCによるフィルタリング
- FactCC [Kry  ́sci ́nski+2020]
  - (Source document, claim sentence) ->  consistent / inconsistent を予測する2値分類モデル
  - ルールベース（entity swap, number swap）等により、claim sentenceの負例を作成

factGSGによる文選択方法
<img width="569" alt="スクリーンショット 2022-08-16 10 36 29" src="https://user-images.githubusercontent.com/13919583/184780203-aac21bfa-7581-498f-9b4a-c129c24c1d6e.png">

### Factuality-Aware Fine-tuning
fine-tuning時にhallucinationが入っていることで、downstream taskに影響を与えてしまっている。
解決策として、３つのコンポーネントを提案する。
- Connector: finetuning時にも[mask]トークンを導入する。maskのbest position についてはappendixを参照されたい。
- Corrector: ソース文書に現れないentityをrefernceから削除／置換する
  - replace: ソース文書とreferenceでfull name / first or last nameのみのgapがある場合がある。その場合はsummaryから first or last nameを削除して表記を合わせる処理を行う。（ただしNERにより同一ラベルと判定されたentityに限る）
  - remove: referenceからhalllucinated entityを削除する。単に削除すると文法が崩れる問題もあるので、dependency arcsも考慮してentityに関連する語も消す
  - combined: replace -> remove を行う
- Contrastor: ソース文書をアンカー、reference summaryはpositive sample、擬似的なsummary（entityをランダムな名前に変える; ただし同じNE labelに限る）をnegative sampleとして、contrastive learningをする
   - extrinsic / intrinsic hallucinationの両方を検証する

<img width="572" alt="スクリーンショット 2022-08-16 11 06 11" src="https://user-images.githubusercontent.com/13919583/184783103-4463c127-3aeb-47b7-b878-5d576b501454.png">

## 結果

自動評価結果
PEGASUS*は、自分たちで学習したモデルであることを示す
DAE／CLIFFはfactuality-awareな比較手法
<img width="566" alt="スクリーンショット 2022-08-16 11 18 06" src="https://user-images.githubusercontent.com/13919583/184784358-34ffeeba-a903-410d-aa84-f4151fd98b76.png">

--- 

人手評価結果：詳細はappendixを参照されたい
<img width="551" alt="スクリーンショット 2022-08-16 11 20 31" src="https://user-images.githubusercontent.com/13919583/184784599-e152e5be-6016-4a2c-9242-a42989639b5a.png">

--- 

文選択手法の比較、FactCC+Rouge1が最も良かった（relative effect）

<img width="580" alt="スクリーンショット 2022-08-16 11 21 16" src="https://user-images.githubusercontent.com/13919583/184784687-ec662424-69d7-4f83-aaac-0a01883831a4.png">

--- 

fine-tuningコンポーネントのablation study

<img width="552" alt="スクリーンショット 2022-08-16 11 25 22" src="https://user-images.githubusercontent.com/13919583/184785102-6d93b82d-088f-43d5-a998-8c9959975dc9.png">

--- 

zero-shotの結果（fine-tuning無しだが、connectorは導入する）

<img width="604" alt="スクリーンショット 2022-08-16 11 26 19" src="https://user-images.githubusercontent.com/13919583/184785203-ba2ceb02-be88-4844-b269-b0eba2feb9d8.png">

--- 

Zero-shot and few-shotの比較

-> PEGASUS, FactGSG+maskは、サンプル10でFactCCスコアが劣化しているが、FactPEGASUSでは改善している
fine-tuning時にcorrector／constractorによりfactualityを考慮できているからではないか。

<img width="1184" alt="スクリーンショット 2022-08-16 11 28 27" src="https://user-images.githubusercontent.com/13919583/184785423-2c7d1705-878d-45cf-9ded-a28265481a8d.png">

--- 

fine-tuningが進むほど、factualityはdegrationするか？
-> FactPEGASUSもdegrationしちえるが、BART-baseと比べると影響は小さい

<img width="1155" alt="スクリーンショット 2022-08-16 11 35 59" src="https://user-images.githubusercontent.com/13919583/184786258-fac3eec7-3506-4163-b66c-f285001cb5c8.png">

---

faithfulness-abstractivenessのトレードオフカーブ

-> extractivenessがやたら上がっているというわけではない
※Selectorは提案した文選択器

<img width="575" alt="スクリーンショット 2022-08-16 11 37 05" src="https://user-images.githubusercontent.com/13919583/184786392-47e2e99f-756b-40ac-8c69-b19fc6de2f60.png">

**Comment by ichiroex** [comment]
## Appendix

Connectorにおいて、Maskトークンをどこに置けばよいか？
-> ソース文書の先頭が最も良かった

<img width="1155" alt="スクリーンショット 2022-08-16 11 41 56" src="https://user-images.githubusercontent.com/13919583/184786973-5522d0c3-90d1-4575-9e1a-db6b9b309334.png">
